Wikipedia/Images/WMFImageController.swift

```diff
@@ -124,14 +123,54 @@ public class WMFImageController : NSObject {
         .then(mainImageBlock)
     }
 
-    /// MARK: - Simple Fetching
+    /**
+    Use the shared session to download an image directly to the receiver's disk cache.
+
+    - parameter url: The URL to download the image from.
+
+    - returns: A promise which is resolved after the data has been successfully written to disk.
+    */
+    public func downloadImageDataInBackground(url: NSURL) -> Promise<Void> {
```

Comment:
Wondering why doesn't the method and `fetchImageWithURL` method call through to the same logic? The only thing that should be different is the priority and the cache destinations - which are all configurable using SDWebImage's Options instead of the defaults:

`SDWebImageContinueInBackground`
`SDWebImageCacheMemoryOnly`
`SDWebImageLowPriority`